File: boj_13077.cs
using System;
using System.Text;
using System.Collections.Generic;
class Program
{
  static void Main(String[] args)
  {
    StringBuilder sb = new StringBuilder();
    int t = int.Parse(Console.ReadLine());
    while (t-- > 0)
    {
      String[] str = Console.ReadLine().Split('/');
      int p = int.Parse(str[0]);
      int q = int.Parse(str[1]);
      int idx = 1;
      Stack<int> stack = new Stack<int>();
      while (true)
      {
        if (p == 1 && q == 1) break;
        if (p > q)
        {
          stack.Push(1);
          p -= q;

        }
        else
        {
          stack.Push(0);
          q -= p;
        }
      }
      while (stack.Count != 0)
      {
        if (stack.Pop() == 0)
        {
          idx = idx * 2;
        }
        else
        {
          idx = idx * 2 + 1;
        }
      }
      sb.Append(idx + "\n");
    }
    Console.WriteLine(sb);
  }
}

File: boj_13077.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;
import java.util.StringTokenizer;

public class Main {

    public static void main(String args[]) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int t = Integer.parseInt(br.readLine());
        while (t-- > 0) {
            StringTokenizer st = new StringTokenizer(br.readLine(), "/");
            int p = Integer.parseInt(st.nextToken());
            int q = Integer.parseInt(st.nextToken());
            int idx = 1;
            Stack<Integer> stack = new Stack<>();
            while (true) {
                if (p == 1 && q == 1) break;
                if (p > q) {
                    stack.add(1);
                    p -= q;

                } else {
                    stack.add(0);
                    q -= p;
                }
            }
            while (!stack.isEmpty()) {
                if (stack.pop() == 0) {
                    idx = idx * 2;
                } else {
                    idx = idx * 2 + 1;
                }
            }
            sb.append(idx + "\n");
        }
        System.out.println(sb);
    }
}
